book/src/building/swift-package/README.md

```diff
@@ -0,0 +1,221 @@
+swift-bridge-build = "0.1"
+
+[dependencies]
+swift-bridge = "0.1"
+```
+
+In `lib.rs`, add the following:
+
+```rust
+// my_rust_lib/src/lib.rs
+#[swift_bridge::bridge]
+mod ffi {
+    extern "Rust" {
+        fn hello_rust();
+    }
+}
+
+fn hello_rust() -> String {
+    String::from("Hello Rust!")
+}
+```
+
+Add a new `build.rs` file:
+```rust
+// my_rust_lib/build.rs
+
+use std::path::PathBuf;
+
+fn main() {
+    let out_dir = PathBuf::from("./generated");
+
+    let bridges = vec!["src/lib.rs"];
+    for path in &bridges {
+        println!("cargo:rerun-if-changed={}", path);
+    }
+
+    swift_bridge_build::parse_bridges(bridges)
+        .write_all_concatenated(out_dir, env!("CARGO_PKG_NAME"));
+}
+```
+
+Build the project for the desired platforms:
+
+```bash
+cargo build --target x86_64-apple-darwin
+cargo build --target aarch64-apple-ios
+cargo build --target x86_64-apple-ios
+```
+
+## Creating the XCFramework
+
+Go back to the root of the project and make a new directory.
+
+```bash
+mkdir MyFramework && cd $_
+```
+
+Copy the generated libraries and the headers to this folder:
+
+```
+MyFramework
+├── include
+│   ├── SwiftBridgeCore.h
+│   ├── module.modulemap
+│   └── my_rust_lib.h
+├── ios
+│   └── libmy_rust_lib.a
+├── macos
+│   └── libmy_rust_lib.a
+└── simulator
+    └── libmy_rust_lib.a
+```
+
+Add `include/module.modulemap`:
+
+```modulemap
+module MyRustLib {
+    header "my_rust_lib.h"
+    header "SwiftBridgeCore.h"
+    export *
+}
+```
+
+Now it is time to build the xcframework:
+
+```bash
+xcodebuild -create-xcframework \
+    -library simulator/libmy_rust_lib.a \
+    -headers include \
+    -library ios/libmy_rust_lib.a \
+    -headers include \
+    -library macos/libmy_rust_lib.a \
+    -headers include \
+    -output MyRustLib.xcframework
+```
+
+*The order of the `library` tags is important*
+
+## Creating the Swift package
+
+Go back to the root of the project and create a new Swift package, copy the xcframework and generated swift files to it:
+
+```
+MySwiftPackage
+├── Sources
+│   └── MySwiftPackage
+│       └── SwiftBridgeCore.swift
+│       └── my_rust_lib.swift
+├── MyRustLib.b.xcframework
+└── Package.swift
+```
+
+Add the framework as a binary target
+```swift
+// MySwiftPackage/Package.swift
+
+let package = Package(
+    name: "MySwiftPackage",
+    products: [
+        .library(
+            name: "MySwiftPackge",
+            targets: ["MySwiftPackge"]),
+    ],
+    dependencies: [
+
+    ],
+    targets: [
+        .binaryTarget(
+            name: "MyRustLib",
+            path: "MyRustLib.xcframework"
+        ),
+        .target(
+            name: "MySwiftPackge",
+            dependencies: ["MyRustLib"]),
+    ]
+)
+```
+
+We will need to import our rust library in the `.swift` files:
+
+```swift
+// MySwiftPackage/Sources/MySwiftPackage/SwiftBridgeCore.swift
+import MyRustLib
+
+// ...
+```
+
+```swift
+// MySwiftPackage/Sources/MySwiftPackage/my_rust_lib.swift
+import MyRustLib
```

Comment:
Is it possible to not touch this generated file, or is this necessary? 
I'd like to avoid touching this file if possible.

i.e. can there just me another file that imports this and imports `MyRustLib` or something like that..?

Reply:
I'm not sure that this is possible. This file needs access to `__swift_bridge__$hello_rust()`, which is a function defined in `MyRustLib`.

Reply:
I think you'd need to handle this in the `Package.swift`.

I think this is explained here https://stackoverflow.com/questions/65220359/add-package-dependency-for-a-binary-target-with-swift-package-manager/65255303#65255303

Where basically you create a wrapper target that depends on the binary framework and the Swift code. Then we'd have the library product expose that wrapper target.

Note that I've never actually made a Swift package so I'm just going off of Google.

I'm also totally cool with you just leaving a TODO comment in the documentation for now and we can figure this out later so that this PR can land.

Thanks!

Reply:
I'm not sure that would work, as the swift package also depends on the binary framework. I can look into it some other day, though.

Reply:
I guess it's technically a bit cleaner if we're explicitly importing the Rust that we're using so.. I suppose this isn't the worst thing.

I just don't like the idea of manually editing a generated file.

But.. all of this is temporary until we automate this.. So.. works for now!